compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -2097,52 +2135,77 @@ impl<'a> Parser<'a> {
             }
         }
 
-        recovered
+        if let Some(recovered) = recovered {
```

Comment:
Nit: there's already one `if let Some(token) = recovered` above, this one can be merged into it.
(Also `recovered` doesn't need to be `mut` anymore.)